Efflux transporters play a critical role in bacterial antibiotic resistance by facilitating the removal of harmful substances. These are classified into five distinct families: ABC, MFS, MATE, RND, and SMR. The significant sequence variability among these families, coupled with insufficient functional annotation, presents considerable challenges for traditional categorization methods.  We hypothesize that integrating of ProtGPT2-generated efflux protein sequences with a multi-window convolutional neural network (mCNN) is proposed to enhance classification accuracy by effectively capturing local motifs and broader evolutionary patterns often overlooked by previous methods. Generative models like ProtGPT2 are effective for this purpose, as they generate a variety of sequence variants that reflect patterns observed in natural efflux families, thereby minimizing issues related to data scarcity.   The proposed GenEfflux framework, unlike alignment-based methods such as HHblits and single-feature CNNs, combines generative sequence expanding with multi-scale evolutionary feature extraction via Position-Specific Scoring Matrices (PSSMs), thereby enhancing the understanding of sequence-function relationships. In comparative evaluations, GenEfflux consistently outperformed the baseline deepEfflux model across all Efflux transporter classes. In Class B, sensitivity increased from 0.5385 to 0.9999, and the Matthews correlation coefficient (MCC) rose from 0.4397 to 0.9327.   In Class C, accuracy improved from 0.8977 to 0.9668, alongside an increase in MCC from 0.7668 to 0.9331. The findings demonstrate that sequences generated by ProtGPT2 possess functional relevance and improve classification effectiveness. GenEfflux suggestions a comprehensive framework for enhancing efflux protein analysis and advancing research on antibiotic resistance.
![Grad-CAM or Cosine Similarity Plot](https://github.com/mhb1717/efflux/blob/main/Group%2080%20(1).png?raw=true)


